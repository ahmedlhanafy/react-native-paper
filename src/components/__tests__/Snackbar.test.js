import * as React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-view';
import Snackbar from '../Snackbar.tsx';

// Make sure any animation finishes before checking the snapshot results
jest.mock('Animated', () => {
  const ActualAnimated = jest.requireActual('Animated');

  return {
    ...ActualAnimated,
    timing: (value, config) => ({
      start: callback => {
        value.setValue(config.toValue);
        callback && callback({ finished: true });
      },
    }),
  };
});

it('renders snackbar with content', () => {
  const tree = renderer
    .create(
      <SafeAreaProvider>
        <Snackbar visible onDismiss={jest.fn()}>
          Snackbar content
        </Snackbar>
      </SafeAreaProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders not visible snackbar with content wrapper but no actual content', () => {
  const tree = renderer
    .create(
      <SafeAreaProvider>
        <Snackbar visible={false} onDismiss={jest.fn()}>
          Snackbar content
        </Snackbar>
      </SafeAreaProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders snackbar with Text as a child', () => {
  const tree = renderer
    .create(
      <SafeAreaProvider>
        <Snackbar visible onDismiss={jest.fn()}>
          <Text>Snackbar content</Text>
        </Snackbar>
      </SafeAreaProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders snackbar with action button', () => {
  const tree = renderer
    .create(
      <SafeAreaProvider>
        <Snackbar
          visible
          onDismiss={() => {}}
          action={{ label: 'Undo', onPress: jest.fn() }}
        >
          Snackbar content
        </Snackbar>
      </SafeAreaProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
