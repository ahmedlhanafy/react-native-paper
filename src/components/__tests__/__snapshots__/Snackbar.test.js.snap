// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders not visible snackbar with content wrapper but no actual content 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`renders snackbar with Text as a child 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`renders snackbar with action button 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`renders snackbar with content 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
